### ISSUE TITLE
Add `{post_type}sConnectionOrderbyEnum` to replace `PostObjectsConnectionOrderbyEnum `

### ISSUE THREAD

**Issue by hsimah** [wp-graphql/wp-graphql#917]
As per #728 I feel we should be allowing more granular control over order by inputs.

When a user registers a type they should get a custom order by enum input arg. Users can then use `graphql_{post_type}sConnectionOrderbyEnum_values` to customise the order by fields. (And then `graphql_post_object_connection_query_args` to actually use their fields.)

**Comment by CodeProKid** [comment]
I'm not _totally_ convinced that this is needed. Would be great to see a use case to support this need.

**Comment by hsimah** [comment]
I think the use case was something like ordering on custom fields:

Given a collection of `Quizzes` I want to retrieve pages ordered by their `difficulty`. I can use `graphql_post_object_connection_query_args` to correctly do my server side ordering on the WP_Query.

I could add in my own order by input, and this turned out to be somewhat larger than I anticipated when I started on the PR.

**Comment by jasonbahl** [comment]
Ya, to be able to express what field(s) can be ordered by, you want to expose just what's possible for any given connection. 

Right now you'd have to add `difficulty` to the existing Enum, and that will show on _all_ connections for any Post Type, which makes the self-documenting Schema funky. 

Like why should the docs for Posts show that I could sort by "difficulty" when I really can't because "difficulty" is _only_ a property of "Quizzes" post type. 

This might be a larger discussion on filtering connections in general though

**Comment by hsimah** [comment]
Yeah, that was what I found when I was working on this issue - the order by is reused a lot. 

But I still would like a central, non-janky way of customizing order by inputs for my types. It seems lame to have to add my own order by input for each type. Perhaps we could have an override in the register config to add custom order by. It's been a while since I looked at this.

tl;dr I agree that this issue does not have a good solution, but I do want to get custom order by somehow.
